### PR TITLE
ACS-8329 Manage bulks endpoint

### PR DESCRIPTION
**Pull request by tomson7777**
JIRA TICKET:

https://hyland.atlassian.net/browse/ACS-8329

**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Manage bulk hold endpoint was not implemented

**What is the new behaviour?**
Manage bulk hold endpoint has been added and documented.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

### CODE REVIEW

File: lib/js-api/src/api/gs-core-rest-api/docs/LegalHoldApi.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ All URIs are relative to _https://localhost/alfresco/api/-default-/public/gs/ver
 | [**assignHold**](LegalHoldApi.md#assignHold)     | **POST** /holds/{holdId}/children            | Assign node to legal hold     |
 | [**assignHolds**](LegalHoldApi.md#assignHolds)   | **POST** /holds/{holdId}/children            | Assign nodes to legal hold    |
 | [**unassignHold**](LegalHoldApi.md#unassignHold) | **DELETE** /holds/{holdId}/children/{nodeId} | Unassign node from legal hold |
-[**createHold**](LegalHoldApi.md#createHold) | **POST** /file-plans/{filePlanId}/holds | Create one hold
-[**createHolds**](LegalHoldApi.md#createHolds) | **POST** /file-plans/{filePlanId}/holds | Create list of holds
+| [**createHold**](LegalHoldApi.md#createHold)     | **POST** /file-plans/{filePlanId}/holds      | Create one hold               |
+| [**createHolds**](LegalHoldApi.md#createHolds)   | **POST** /file-plans/{filePlanId}/holds      | Create list of holds          |
```
Comment (DaryaBalvanovich) [inline lib/js-api/src/api/gs-core-rest-api/docs/LegalHoldApi.md:12]: seems you miss to add `bulkHold` to the table

File: lib/js-api/src/api/gs-core-rest-api/docs/LegalHoldApi.md [hunk 2]
```diff
@@ -257,21 +263,66 @@ let opts = [
...
+
+[**HoldPaging**](./HoldPaging.md)
+
+<a name="bulkHold"></a>
+
+# **bulkHold**
+
+> BulkHoldAddResponse bulkHold(holdId, query, language)
+
+Start the asynchronous bulk process for a hold with id holdId based on search query results.
```
Comment (DaryaBalvanovich) [inline lib/js-api/src/api/gs-core-rest-api/docs/LegalHoldApi.md:293]:
```suggestion
Start the asynchronous bulk process for a hold with id `holdId` based on search query results.
```

File: lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts [hunk 3]
```diff
@@ -153,4 +154,30 @@ export class LegalHoldApi extends BaseApi {
             returnType: HoldPaging
         });
     }
+
+    /**
+     * Start the asynchronous bulk process for a hold with id holdId based on search query results.
+     *
+     * @param holdId The identifier of a hold
+     * @param query search query
+     * @param language language
```
Comment (DaryaBalvanovich) [inline lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts:163]:
```suggestion
     * @param query Search query
     * @param language Language code
```

File: lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts [hunk 4]
```diff
@@ -105,4 +105,16 @@ export class LegalHoldService {
...
         return from(this.legalHoldApi.createHolds(filePlanId, holds));
     }
+
+    /**
+     * Start the asynchronous bulk process for a hold with id holdId based on search query results.
+     *
+     * @param holdId The identifier of a hold
+     * @param query search query
+     * @param language language
+     * @returns Promise<BulkHoldAddResponse>
```
Comment (DaryaBalvanovich) [inline lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts:115]:
```suggestion
     * @returns Observable <BulkHoldAddResponse>
```

**Review by DaryaBalvanovich — APPROVED** [review]

**Comment by DenysVuika** [comment]
needs rebase

**Review by g-jaskowski — APPROVED** [review]

**Pull request merged**